src/lib/whenIdle.ts

```diff
@@ -17,7 +17,10 @@
 import {onHidden} from './onHidden.js';
 import {runOnce} from './runOnce.js';
 
-const rIC = self.requestIdleCallback || self.setTimeout;
+// Used by unit test so expose via globalThis with fallback for unsupporting browsers
+const rIC = globalThis
```

Comment:
Rather than using `globalThis` (which will error in some older browsers), I think it would be better to keep this line as it was but move it within the `whenIdle()` function (to line 27 in the original file).

This is similar to what we do for cases that reference `document` in other parts of the code.

Reply:
Ah clever. That seems to work. Done.